sulfur_simulation/show_simulation.py

```diff
@@ -53,6 +69,70 @@ def update(frame: int) -> tuple[PathCollection]:
     )
 
 
+def animate_particle_positions_hexagonal(
```

Comment:
could you make this geenric in terms of delta_x and delta_y as two length 2 vectors?